pom.xml

```diff
@@ -278,8 +276,6 @@
                 <goal>report</goal>
               </goals>
               <configuration>
-                <!-- Sets the path to the file which contains the execution data. -->
-                <dataFile>${project.build.directory}/coverage-reports/jacoco-ut.exec</dataFile>
```

Comment:
As per the following link, you can have JaCoCo append coverage of all modules in the same file in the parent module target folder. We can change the dataFile to "${project.basedir}/../target/coverage-reports/jacoco-ut.exec".

https://community.sonarsource.com/t/unit-tests-across-multi-module-projects/9662/3

Reply:
1. This seems to be specific to how sonarqube works and we don't use sonar/sonarqube for aggregation
2. [This](https://github.com/jacoco/jacoco/wiki/MavenMultiModule#usage) is the recommendation from official jacoco documentation, which I have followed in this PR.

I did try out the sonar documentation on a test project locally and couldn't make it to work.